test/snippet/search/bi_fm_index.cpp

```diff
@@ -15,7 +15,7 @@ int main()
     cur.extend_left("AA"_dna4);                                         // search the pattern "AAGG"
     seqan3::debug_stream << "Number of hits: " << cur.count() << '\n';  // outputs: 2
     seqan3::debug_stream << "Positions in the genome: ";
-    for (auto const & pos : cur.locate())                               // outputs: 8, 22
+    for (auto const & pos : cur.locate())                               // outputs: (0, 8), (0, 22)
```

Comment:
hm.. I'm not sure :woman_shrugging:

Reply:
In general, in a generic context we need to promote `auto &&` since this encapsualtes true type deduction. In non-generic cases `const &` would be preferred since this is more explicit on what I can do on the type. Later it would be something like `std::integral auto && pos`. So since this is a generic case, we should promote the `auto &&` version.

Reply:
and in the other places below